massa-network-worker/src/messages.rs

```diff
@@ -147,10 +150,22 @@ impl SerializeCompact for Message {
                     )
                 })?;
                 res.extend(list_len.to_varint_bytes());
-                for hash in list {
+                for (hash, info) in list {
                     res.extend(hash.to_bytes());
+                    let info_type = match info {
+                        AskForBlocksInfo::Info => BlockInfoType::Info,
+                        AskForBlocksInfo::Operations(_) => BlockInfoType::Operations,
+                    };
+                    res.extend(u32::from(info_type).to_varint_bytes());
+                    if let AskForBlocksInfo::Operations(ids) = info {
+                        let op_ids_serializer = OperationIdsSerializer::new();
+                        op_ids_serializer.serialize(ids, &mut res)?;
+                    }
                 }
             }
+            Message::ReplyForBlocks(list) => {
+                panic!("Message::ReplyForBlocks should be constructed using shared storage.");
```

Comment:
You are using the old system of storage. With #2772 you will be able to pass storage to the serializer

Reply:
Ok, thanks for the info.

Reply:
Can you perhaps make that change?